src/client.ts

```diff
@@ -1642,51 +1659,64 @@ export class Client<
 	 */
 	private async fetch<T = unknown>(
 		url: string,
-		// TODO: Change to `params` when Authorization header support works in browsers with CORS.
-		// _params?: Partial<BuildQueryURLArgs>,
 		params: FetchParams = {},
 	): Promise<T> {
-		// TODO: Restore when Authorization header support works in browsers with CORS.
-		// const accessToken = (params && params.accessToken) || this.accessToken;
-		// const options = accessToken
-		// 	? { headers: { Authorization: `Token ${accessToken}` } }
-		// 	: {};
+		let job: Promise<FetchJobResult>;
 
-		const res = await this.fetchFn(url, {
-			signal: params.signal,
-		});
+		if (this.fetchJobs[url] && this.fetchJobs[url].has(params.signal)) {
```

Comment:
Maybe it's worth adding some comments here explaining why we're doing that and/or a link to this PR

Reply:
Done!

```
		// `fetchJobs` is keyed twice: first by the URL and again by is
		// signal, if one exists.
		//
		// Using two keys allows us to reuse fetch requests for
		// equivalent URLs, but eject when we detect unique signals.
```